Core/MSBuild/MSBuild.cs

```diff
@@ -46,22 +46,21 @@ public void CallMSBuild (string msBuildPath, string arguments)
 
     var msBuildStartInfo = new ProcessStartInfo(msBuildPath, arguments);
     msBuildStartInfo.UseShellExecute = false;
-    msBuildStartInfo.RedirectStandardError = true;
     msBuildStartInfo.RedirectStandardOutput = true;
     msBuildStartInfo.WorkingDirectory = Environment.CurrentDirectory;
 
     var msBuildProcess = Process.Start(msBuildStartInfo);
+
+    var msBuildOutput = msBuildProcess!.StandardOutput.ReadToEnd();
+    
     msBuildProcess!.WaitForExit(c_msBuildProcessTimeout);
```

Comment:
The deadlock fix implies we have to read before the process exits? @a-ctor how does one deal with getting all the output until the process has exited?

Reply:
What do you mean? We do the `StandardOutput.ReadToEnd()` first, which will block until standard out is closed and then we `.WaitForExit()`. 

The deadlock came from the fact that standard output has a buffer, which if full will block the application writing to it. If we do `.WaitForExit()` first then we are not clearing the buffer and if the application writes to much it will block causing the deadlock.

Reply:
Oh, I.C. Thanks. => we definitely need a short explainatory comment in the code that we need to first read then exit and why it is correct. These details are not self-evident :)